nifi-nar-bundles/nifi-adx-bundle/nifi-adx-processors/src/test/java/org/apache/nifi/processors/adx/AzureAdxIngestProcessorTest.java

```diff
@@ -0,0 +1,309 @@
+
+        //missing ingest url required for connection service
+        testRunner.setProperty(azureAdxConnectionService,AzureAdxConnectionService.APP_ID,"sample");
+        testRunner.setProperty(azureAdxConnectionService,AzureAdxConnectionService.APP_KEY,"sample");
+        testRunner.setProperty(azureAdxConnectionService,AzureAdxConnectionService.APP_TENANT,"sample");
+
+        IllegalStateException illegalStateException =Assertions.assertThrows(IllegalStateException.class, () -> {
+            testRunner.enableControllerService(azureAdxConnectionService);
+        });
+
+        assertTrue(illegalStateException.getMessage().contains("Ingest URL"));
+        assertTrue(illegalStateException.getMessage().contains("invalid"));
+
+    }
+
+    @Test
+    public void testIngestionFailure() throws InitializationException, IngestionClientException, IngestionServiceException {
+
+        testRunner.setProperty(AzureAdxIngestProcessor.TABLE_NAME,MOCK_TABLE_NAME);
+        testRunner.setProperty(AzureAdxIngestProcessor.IM_KIND,AzureAdxIngestProcessor.IM_KIND_CSV);
+        testRunner.setProperty(AzureAdxIngestProcessor.DB_NAME,MOCK_DB_NAME);
+        testRunner.setProperty(AzureAdxIngestProcessor.MAPPING_NAME,MOCK_MAPPING_NAME);
+        testRunner.setProperty(AzureAdxIngestProcessor.DATA_FORMAT,"CSV");
+        testRunner.setProperty(AzureAdxIngestProcessor.IR_LEVEL,"IRL_FAS");
+        testRunner.setProperty(AzureAdxIngestProcessor.WAIT_FOR_STATUS,"ST_SUCCESS");
+        testRunner.setProperty(AzureAdxIngestProcessor.IR_METHOD,"IRM_TABLE");
+        testRunner.setProperty(AzureAdxIngestProcessor.ADX_SERVICE,"adx-connection-service");
+
+
+        testRunner.setValidateExpressionUsage(false);
+
+        azureAdxConnectionService = new AzureAdxConnectionService();
+
+        testRunner.addControllerService("adx-connection-service", azureAdxConnectionService, new HashMap<>());
+
+        testRunner.setProperty(azureAdxConnectionService,AzureAdxConnectionService.INGEST_URL,"http://ingest-sample.com/");
+        testRunner.setProperty(azureAdxConnectionService,AzureAdxConnectionService.APP_ID,"sample");
+        testRunner.setProperty(azureAdxConnectionService,AzureAdxConnectionService.APP_KEY,"sample");
+        testRunner.setProperty(azureAdxConnectionService,AzureAdxConnectionService.APP_TENANT,"sample");
+
+        testRunner.enableControllerService(azureAdxConnectionService);
+        testRunner.assertValid(azureAdxConnectionService);
+
+
+        AdxConnectionService adxConnectionServiceFromProcessContext = (AdxConnectionService) testRunner.getProcessContext().getControllerServiceLookup().getControllerService("adx-connection-service");
+        assertNotNull(adxConnectionServiceFromProcessContext);
+
+        InputStream stream = new ByteArrayInputStream("exampleString".getBytes(StandardCharsets.UTF_8));
+        testRunner.enqueue(stream);
+        try {
+            testRunner.run(1);
```

Comment:
There should be tests for each format we support CSV , TSV , AVRO etc etc.  They do not exist in this. 

Please add end to end tests that can optionally be disabled. The tests are inadequate